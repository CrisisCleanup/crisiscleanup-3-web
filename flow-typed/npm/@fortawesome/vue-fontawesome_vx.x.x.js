// flow-typed signature: 88a7e004dad0f95b883e0a7cec99db76
// flow-typed version: <<STUB>>/@fortawesome/vue-fontawesome_v^0.1.10/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/vue-fontawesome'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/vue-fontawesome' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/index.es' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/rollup.config' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/icons' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/converter' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/logger' {
  declare module.exports: any;
}

declare module '@fortawesome/vue-fontawesome/src/utils' {
  declare module.exports: any;
}

// Filename aliases
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/build'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/check-versions'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/utils'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/vue-loader.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.base.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.dev.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/build/webpack.prod.conf'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/dev.env'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/config/prod.env'>;
}
declare module '@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/examples/vue-cli-webpack/src/main'>;
}
declare module '@fortawesome/vue-fontawesome/index.es.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/index.es'>;
}
declare module '@fortawesome/vue-fontawesome/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome'>;
}
declare module '@fortawesome/vue-fontawesome/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome'>;
}
declare module '@fortawesome/vue-fontawesome/rollup.config.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/rollup.config'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__fixtures__/helpers'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__fixtures__/icons.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__fixtures__/icons'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeIcon.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayers.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/__tests__/FontAwesomeLayersText.test'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeIcon'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeLayers'>;
}
declare module '@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/components/FontAwesomeLayersText'>;
}
declare module '@fortawesome/vue-fontawesome/src/converter.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/converter'>;
}
declare module '@fortawesome/vue-fontawesome/src/index' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src'>;
}
declare module '@fortawesome/vue-fontawesome/src/index.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src'>;
}
declare module '@fortawesome/vue-fontawesome/src/logger.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/logger'>;
}
declare module '@fortawesome/vue-fontawesome/src/utils.js' {
  declare module.exports: $Exports<'@fortawesome/vue-fontawesome/src/utils'>;
}
