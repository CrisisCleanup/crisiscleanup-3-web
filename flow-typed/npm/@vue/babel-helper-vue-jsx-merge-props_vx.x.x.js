// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
