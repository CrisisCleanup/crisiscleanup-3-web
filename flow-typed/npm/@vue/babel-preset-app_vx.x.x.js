// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
