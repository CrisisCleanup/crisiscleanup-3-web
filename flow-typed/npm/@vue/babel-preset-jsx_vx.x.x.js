// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
