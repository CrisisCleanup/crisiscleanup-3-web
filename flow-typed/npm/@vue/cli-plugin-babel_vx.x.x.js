// flow-typed signature: 4cf1a98e80b05c0a167e013d64bc9645
// flow-typed version: <<STUB>>/@vue/cli-plugin-babel_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-babel'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-babel' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/usePluginPreset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/preset' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/usePluginPreset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/usePluginPreset'>;
}
declare module '@vue/cli-plugin-babel/generator.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/generator'>;
}
declare module '@vue/cli-plugin-babel/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/preset'>;
}
