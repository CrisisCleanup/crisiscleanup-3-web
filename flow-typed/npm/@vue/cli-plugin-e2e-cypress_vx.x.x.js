// flow-typed signature: 122252e4d027555936ded476aff636b4
// flow-typed version: <<STUB>>/@vue/cli-plugin-e2e-cypress_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-e2e-cypress'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-e2e-cypress' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-e2e-cypress/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-e2e-cypress/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/ui'>;
}
