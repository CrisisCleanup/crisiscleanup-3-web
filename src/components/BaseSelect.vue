<template>
  <div>
    <Multiselect
      ref="input"
      :model-value="modelValue"
      :required="required"
      :options="options"
      :can-clear="clearable"
      :searchable="searchable"
      :label="label"
      :value-prop="itemKey"
      :limit="limit"
      :mode="multiple ? 'tags' : 'single'"
      :disabled="disabled"
      class="form-select text-base"
      :resolve-on-load="false"
      :delay="typeof options === 'function' ? 0 : undefined"
      :filter-results="typeof options === 'function' ? false : undefined"
      :class="[
        isInvalid && !modelValue ? 'invalid' : '',
        floatLabel ? 'py-2 pt-3' : '',
      ]"
      :placeholder="placeholder"
      :no-options-text="placeholder"
      :classes="multiSelectClasses"
      v-bind="$attrs"
      @update:modelValue="
        (v) => {
          onInput(v);
        }
      "
    >
      <template #singlelabel="{ value }">
        <slot name="selected-option" :option="value" />
      </template>

      <template #option="{ option }">
        <slot name="option" :option="option" />
      </template>

      <template #beforelist>
        <slot name="list-header" />
      </template>
    </Multiselect>
    <label
      v-if="floatLabel"
      ref="inputLabel"
      :style="isFloated && floatStyle"
      for="select-id"
    >
      <slot name="float-label" v-bind="{ isFloated }">
        {{ floatLabel }}
      </slot>
    </label>
  </div>
</template>

<script>
import Multiselect from '@vueform/multiselect';
import { computed, h, nextTick, onMounted, ref } from 'vue';
import { useI18n } from 'vue-i18n';
import { isEmpty as empty, kebabCase, xor } from 'lodash';

export default {
  name: 'BaseSelect',
  components: {
    Multiselect,
  },
  props: {
    searchable: {
      type: Boolean,
      default: false,
    },
    multiple: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    required: {
      type: Boolean,
      default: false,
    },
    clearable: {
      type: Boolean,
      default: true,
    },
    selectClasses: {
      type: String,
      default: '',
    },
    containerClasses: {
      type: String,
      default:
        'border relative mx-auto w-full flex items-center justify-end cursor-pointer bg-white text-base leading-snug outline-none',
    },
    options: {
      type: [Array, Function],
      default: () => {
        return [];
      },
    },
    limit: {
      type: Number,
      default: 0,
    },
    placeholder: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: 'large',
    },
    modelValue: {
      type: null,
      default: null,
    },
    itemKey: {
      type: null,
      default: null,
    },
    indicatorIcon: {
      type: String,
      default: 'sort',
    },
    floatLabel: {
      type: String,
      default: '',
    },
  },
  setup(props, { emit }) {
    const { t } = useI18n();
    const cancelText = t('actions.cancel');

    const iconSize = props.multiple ? 'xxs' : 'medium';
    const { indicatorIcon } = props;
    const input = ref(null);
    const inputLabel = ref(null);
    const isFloated_ = ref(false);
    const isEmpty_ = ref(false);
    const isInvalid = ref(false);
    const baseHeight_ = ref(0);
    const currentHeight_ = ref(0);
    const floatDisplacement = ref(0);
    const selected = ref(null);

    const inputIdSelector = computed(() => {
      const idSpec = props.floatLabel ? props.floatLabel : '';
      return `select-id-${kebabCase(idSpec)}`;
    });

    const inputRef = computed(() => {
      return input.value || document.getElementById(inputIdSelector.value);
    });

    const heightMultiplier = computed(() => {
      if (currentHeight_.value === 0) return 1;
      return currentHeight_.value / baseHeight_.value;
    });

    const floatStyle = computed(() => {
      const displace = heightMultiplier.value * -12;
      return { transform: `translateY(${displace}px)` };
    });

    const isEmpty = computed(() => {
      if (props.modelValue !== null) {
        return empty(props.modelValue);
      }
      return isEmpty_.value;
    });

    const isFloated = computed(() => {
      if (isFloated_.value === true) {
        return true;
      }
      if (isEmpty.value === false) {
        return true;
      }
      return isFloated_.value;
    });

    const multiSelectClasses = computed(() => {
      // only override the default classes if we have a custom class
      const searchClassesObj = props.selectClasses
        ? { search: props.selectClasses }
        : {};
      const containerClassesObj = props.containerClasses
        ? { container: props.containerClasses }
        : {};
      return {
        ...searchClassesObj,
        ...containerClassesObj,
        optionSelected: 'text-white bg-crisiscleanup-dark-200',
        optionPointed: 'text-gray-800 bg-crisiscleanup-dark-100',
        optionSelectedPointed:
          'text-white bg-crisiscleanup-dark-200 opacity-90',
        tag: 'text-xs bg-white py-0.5 pl-2 rounded mr-1 mb-1 flex items-center whitespace-nowrap rtl:pl-0 rtl:pr-2 rtl:mr-0 rtl:ml-1 border border-crisiscleanup-dark-100',
      };
    });

    function onInput(value) {
      isEmpty_.value = value === null;
      if (value === null && props.floatLabel) {
        isFloated_.value = false;
        inputLabel.value.classList.remove('focused');
      }
      if (props.multiple) {
        const current = props.modelValue || [];
        const currentValues = current
          .map((item) => (props.itemKey ? item[props.itemKey] : item))
          .filter(Boolean);
        emit('changed', xor(value, currentValues));
      }
      if (props.required && inputRef.value.checkValidity()) {
        isInvalid.value = false;
      }
    }

    function onBlur() {
      if (props.floatLabel && isEmpty.value) {
        isFloated_.value = false;
        inputLabel.value.classList.remove('focused');
      }
    }

    function open() {
      if (props.floatLabel) {
        isFloated_.value = true;
        inputLabel.value.classList.add('focused');
      }
      nextTick(() => {
        const items = Array.prototype.slice.call(
          document.querySelectorAll('.vs__dropdown-option'),
        );

        for (const item of items) {
          item.classList.remove('vs__dropdown-option--highlight');
        }

        const selected = items.find((item) => {
          if (props.modelValue) {
            const currentLabel = props.label
              ? props.modelValue[props.label]
              : props.modelValue;
            return item.textContent.trim() === currentLabel;
          }
          return false;
        });

        if (selected) {
          selected.classList.add('vs__dropdown-option--highlight');
          document.querySelector('.vs__dropdown-menu').scrollTop =
            selected.offsetTop;
        }
      });
    }
    function handleResize() {
      currentHeight_.value = inputRef.value.parentElement.clientHeight;
      emit('resize', currentHeight_.value);
    }

    onMounted(() => {
      if (props.required) {
        inputRef.value.addEventListener(
          'invalid',
          () => {
            isInvalid.value = true;
          },
          true,
        );
      }
      if (props.floatLabel) {
        inputRef.value.parentElement.classList.add('has-float');
        baseHeight_.value = inputRef.value.parentElement.clientHeight;
      }
    });

    return {
      selected,
      // options,
      isInvalid,
      floatDisplacement,
      cancelText,
      inputIdSelector,
      inputRef,
      heightMultiplier,
      floatStyle,
      isEmpty,
      isFloated,
      multiSelectClasses,
      onInput,
      onBlur,
      open,
      handleResize,
      input,
      inputLabel,
    };
  },
};
</script>

<style src="@vueform/multiselect/themes/tailwind.css"></style>
