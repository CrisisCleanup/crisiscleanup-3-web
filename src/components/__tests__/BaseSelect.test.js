/**
 *
 * Tests for BaseSelect
 *
 * Components
 */

import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { shallowMount } from "@vue/test-utils";
import BaseSelect from "../BaseSelect";
library.add(fas);

describe("BaseSelect", () => {
  it("should render correctly and match snapshot", () => {
    const testData = {
      propsData: {
        placeholder: "Select Something",
        icon: "caret-down",
        class: "incident-select"
      },
      slots: {
        options: "<div>option</div>"
      },
      stubs: {
        "font-awesome-icon": FontAwesomeIcon
      }
    };
    const wrapper = shallowMount(BaseSelect, testData);
    expect(wrapper).toMatchSnapshot();
  });
});
