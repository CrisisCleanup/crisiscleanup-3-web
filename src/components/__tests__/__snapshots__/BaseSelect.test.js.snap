// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSelect should render correctly and match snapshot 1`] = `
<a-select size="large" class="incident-select">
  <div>option</div> <template></template>
</a-select>
`;
